nrf52-hal-common/src/spim.rs

```diff
@@ -143,7 +141,7 @@ impl<T> Spim<T> where T: SpimExt {
             //
             // The MAXCNT field is 8 bits wide and accepts the full range of
             // values.
-            unsafe { w.maxcnt().bits(tx_buffer.len() as u8) }
+            unsafe { w.maxcnt().bits(tx_buffer.len() as _) }
```

Comment:
`maxcnt` is apparently wider on the 52840, so this casts to whatever the size may be

Reply:
The comment should be updated then. The "8 bits wide" bit is misleading.

Reply:
I've expanded the comment in the latest iteration.  I'll file an issue for handling data longer than 8 bits allow.